frontend/src/components/unsubscribe/tests/Unsubscribe.test.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import { render, mockCommonApis, server, screen } from 'test-utils'
+import Unsubscribe from '../Unsubscribe'
+import userEvent from '@testing-library/user-event'
+import { Route } from 'react-router-dom'
+
+function mockApis() {
+  const { handlers } = mockCommonApis({
+    campaigns: [
+      {
+        id: 1,
+        name: 'Test email campaign',
+        type: 'EMAIL',
+        created_at: new Date(),
+        valid: true,
+        protect: false,
+        demo_message_limit: null,
+        csv_filename: 'test_csv_filename.csv',
+        is_csv_processing: false,
+        num_recipients: 1,
+        job_queue: [],
+        halted: false,
+      },
+    ],
+  })
+  return handlers
+}
+
+function renderUnsubscribe() {
+  render(
+    <Route exact path="/unsubscribe/:version">
+      <Unsubscribe />
+    </Route>,
+    {
+      router: {
+        initialIndex: 0,
+        initialEntries: ['/unsubscribe/v1?c=1&r=testEmail@open.gov.sg&h=testH'],
+      },
+    }
+  )
+}
+
+test('successfully unsubscribes user from a campaign', async () => {
```

Comment:
It would be good to also include a test on the option where the user chooses not to unsubscribe

Reply:
I've added a test for that. I'm not sure if there's much to test other than to assert that the necessary UI elements are present after clicking the "I prefer to stay" button.

Reply:
Yup that's fine, thought since we asserted that the elements are rendering correctly for the protected mail page we could do that here too.